modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -217,85 +166,91 @@ def get_files_info(ip, _rport, shares, info)
-            subdirs.shift
-            next
-          end
-        elsif depth > datastore['MaxDepth']
+        # Skip user directories if we do not want to spider them
+        if (subdirs.first == USERS_DIR || subdirs.first == DOCUMENTS_DIR) && !datastore['SpiderProfiles']
           subdirs.shift
           next
         end
-        read, write, type, files = eval_host(ip, x, subdirs[0])
-        if files && (read || write)
-          if files.length < 3
-            subdirs.shift
-            next
-          end
-          header = ''
-          if simple.client.default_domain && simple.client.default_name
-            header << " \\\\#{simple.client.default_domain}"
-          end
-          header << "\\#{x.sub('C$', 'C$\\')}" if simple.client.default_name
-          header << subdirs[0]
+        depth = subdirs.first.count('\\')
 
-          pretty_tbl = Rex::Text::Table.new(
-            'Header' => header,
-            'Indent' => 1,
-            'Columns' => [ 'Type', 'Name', 'Created', 'Accessed', 'Written', 'Changed', 'Size' ]
-          )
+        if DEFAULT_SHARES.include?(share_name) && datastore['SpiderProfiles'] && (depth - 2) > datastore['MaxDepth']
+          subdirs.shift
+          next
+        end
+
+        read, _write, _type, files = enum_tree(tree, share, subdirs.first)
 
-          f_types = {
-            1 => 'RO', 2 => 'HIDDEN', 4 => 'SYS', 8 => 'VOL',
-            16 => 'DIR', 32 => 'ARC', 64 => 'DEV', 128 => 'FILE'
-          }
-
-          files.each do |file|
-            next unless file[0] && (file[0] != '.') && (file[0] != '..')
-
-            info = file[1]['info']
-            fa = f_types[file[1]['attr']] # Item type
-            fname = file[0] # Filename
-            tcr = to_unix_time(info[3], info[2]) # Created
-            tac = to_unix_time(info[5], info[4]) # Accessed
-            twr = to_unix_time(info[7], info[6]) # Written
-            tch = to_unix_time(info[9], info[8]) # Changed
-            sz = info[12] + info[13] # Size
-
-            # Filename is too long for the UI table, cut it.
-            fname = "#{fname[0, 35]}..." if fname.length > 35
-
-            # Add subdirectories to list to use if SpiderShare is enabled.
-            if (fa == 'DIR') || (fa.nil? && (sz == 0))
-              subdirs.push(subdirs[0] + '\\' + fname)
-            end
-
-            pretty_tbl << [fa || 'Unknown', fname, tcr, tac, twr, tch, sz]
-            detailed_tbl << [ip.to_s, fa || 'Unknown', x.to_s, subdirs[0] + '\\', fname, tcr, tac, twr, tch, sz]
-            logdata << "#{ip}\\#{x.sub('C$', 'C$\\')}#{subdirs[0]}\\#{fname}\n"
+        if files.nil? || files.empty? || !read
+          subdirs.shift
+          next
+        end
+
+        header = ''
+        pretty_tbl = Rex::Text::Table.new(
+          'Header' => header,
+          'Indent' => 1,
+          'Columns' => [ 'Type', 'Name', 'Created', 'Accessed', 'Written', 'Changed', 'Size' ]
+        )
```

Comment:
This mutation of `header` after passing it to a rex table looks like a code smell, can we move the header creation up higher, or the table generation lower